lib/install/management/create.go

```diff
@@ -211,3 +213,69 @@ func (d *Dispatcher) uploadImages(files map[string]string) error {
 	}
 	return nil
 }
+
+// cleanupAfterCreationFailed cleans up the dangling resource pool for the failed VCH and any bridge network if there is any.
+// The function will not abort and early terminate upon any error during cleanup process. Error details are logged.
+func (d *Dispatcher) cleanupAfterCreationFailed(conf *config.VirtualContainerHostConfigSpec, cleanupNetwork bool) {
+	defer trace.End(trace.Begin("", d.op))
+	var err error
+
+	// clean up dangling empty resource pool
+	d.op.Info("Clean up dangling VCH resource pool after VCH creation failed...")
+
+	if err = d.cleanupEmptyPool(conf); err != nil {
+		d.op.Errorf("Failed to clean up dangling VCH resource pool: %s", err)
+	}
+
+	d.op.Info("Successfully cleaned up dangling resource pool.")
+
+	// only clean up bridge network created if told to
+	if cleanupNetwork {
```

Comment:
If we go with the defer pattern suggested above we could define a _separate_ function and make a deferred call to it once the network has been created.